libdnf/rpm/reldep_splitter.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright (C) 2018-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "reldep_splitter.hpp"
+
+#include <regex>
+#include <string>
+
+namespace libdnf::rpm {
+
+static const std::regex RELDEP_REGEX("^(\\S*)\\s*(<=|>=|<|>|=)?\\s*(\\S*)$");
+
+static bool
+set_cmp_type(Reldep::ComparisonType * cmp_type, std::string cmp_type_string, long int length)
+{
+    const char * cstring = cmp_type_string.c_str();
+
+    if (length == 2) {
+        // The second character is always '=' because of regex
+        if (cstring[0] == '<') {
+            *cmp_type = Reldep::ComparisonType::LT_EQ;
+            return true;
+        } else if (cstring[0] == '>') {
+            *cmp_type = Reldep::ComparisonType::GT_EQ;
+            return true;
+        } else {
+            return false;
+        }
+    } else if (length == 1) {
+        if (cstring[0] == '>') {
+            *cmp_type = Reldep::ComparisonType::GT;
+            return true;
+        } else if (cstring[0] == '<') {
+            *cmp_type = Reldep::ComparisonType::LT;
+            return true;
+        } else if (cstring[0] == '=') {
+            *cmp_type = Reldep::ComparisonType::EQ;
+            return true;
+        } else {
+            return false;
+        }
+    }
+    return false;
+}
+
+bool
+ReldepSplitter::parse(const std::string & reldep_str)
+{
+    enum { NAME = 1, CMP_TYPE = 2, EVR = 3};
+    std::smatch match;
+    if (!std::regex_match(reldep_str, match, RELDEP_REGEX)) {
```

Comment:
any chance to use libsolv's or rpm's internal functions instead?

Reply:
So far I did not find anything in libsolv.

Reply:
don't worry about it then; resolving this conversation